src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyInputBuilderService.java

```diff
@@ -0,0 +1,514 @@
+
+            if (branches.isIstN() || branches.isIstNm1()) {  // Ist activated : we consider the permanent limit
+                // permanent limit on side 1
+                boolean factors1Generated = genFactorForPermanentLimit(branch, Branch.Side.ONE, currentLimits1, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                // permanent limit on side 2
+                boolean factors2Generated = genFactorForPermanentLimit(branch, Branch.Side.TWO, currentLimits2, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                if (!factors1Generated && !factors2Generated) {
+                    // no temporary limit on one side : report and throw exception
+                    reporter.report(Report.builder()
+                        .withKey("monitoredBranchNoPermanentLimits")
+                        .withDefaultMessage("No permanent limits for the monitored branch ${id}")
+                        .withSeverity(TypedValue.ERROR_SEVERITY)
+                        .withValue("id", monitoredEquipment.getId())
+                        .build());
+                    throw new PowsyblException("Branch '" + branch.getId() + "' has no permanent limits !!");
+                }
+            }
+
+            if (StringUtils.isNotEmpty(branches.getLimitNameN())) {  // here we consider the temporary limits
+                // temporary limits on side 1
+                boolean factors1Generated = genFactorForTemporaryLimit(branch, Branch.Side.ONE, currentLimits1, branches.getLimitNameN(), true, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                // temporary limits on side 2
+                boolean factors2Generated = genFactorForTemporaryLimit(branch, Branch.Side.TWO, currentLimits2, branches.getLimitNameN(), true, variableIds, variableSet, sensitivityVariableType, branches, monitoredBranchThreshold, contingencies, result, reporter);
+
+                if (!factors1Generated && !factors2Generated) {
+                    reporter.report(Report.builder()
+                        .withKey("monitoredBranchTemporaryLimitNotFound")
+                        .withDefaultMessage("Temporary limit ${limitName} not found for the monitored branch ${id}")
+                        .withSeverity(TypedValue.ERROR_SEVERITY)
+                        .withValue("limitName", branches.getLimitNameN())
+                        .withValue("id", branch.getId())
+                        .build());
+                    throw new PowsyblException("Temporary limit '" + branches.getLimitNameN() + "' not found for branch '" + branch.getId() + "' !!");
+                }
+            }
+
+            if (StringUtils.isNotEmpty(branches.getLimitNameNm1())) {  // here we consider the temporary limit
```

Comment:
this does not seem to work, if I enable the N-1 IST and give a N-1 limit, then the N limit is taken into while it should not. maybe this should be if (branches.isIstN()) ?

Reply:
Fixed